File: 202. 快乐数/Solution.py
class Solution:
    def getsum(self, n):
        sum = 0
        while n:
            t = n % 10
            sum += t ** 2
            n //= 10
        return sum

    def isHappy(self, n: int) -> bool:
        numbers = set()
        while 1:
            n = self.getsum(n)
            if n == 1:
                return True
            if n in numbers:
                return False
            else:
                numbers.add(n)

File: 202. 快乐数/Solution.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer> numbers = new HashSet<>();
        while(true){
            n = getsum(n);
            if(n == 1) return true;
            if(numbers.contains(n)) return false;
            else numbers.add(n);
        }

    }

    public int getsum(int n){
        int sum = 0;
        while(n != 0){
            int t = n % 10;
            sum += t * t;
            n /= 10;
        }
        return sum;
    }
}